README.md

```diff
@@ -51,23 +51,37 @@ The latest version under the `canary` tag **(latest commit to `main`)** is:
 
 ### Supported TypeScript Version
 
-**The version range of TypeScript currently supported by this parser is `>=3.3.1 <4.9.0`.**
+**The version range of TypeScript currently supported by this parser is `>=4.2.0 <4.9.0`.**
 
-These versions are what we test against.
+Note that we mirror [DefinitelyTyped's version support window](https://github.com/DefinitelyTyped/DefinitelyTyped/#support-window) - meaning we only support versions of TypeScript less than 2 years old.
 
-We will always endeavor to support the latest stable version of TypeScript.
-Sometimes, but not always, changes in TypeScript will not require breaking changes in this project, and so we are able to support more than one version of TypeScript.
-In some cases, we may even be able to support additional pre-releases (i.e. betas and release candidates) of TypeScript, but only if doing so does not require us to compromise on support for the latest stable version.
+You may find that our tooling works on older TypeScript versions however we provide no guarantees and **_we will not accept issues against unsupported versions_**.
+
+#### Supporting New TypeScript Releases
+
+With each new TypeScript release we file an issue to track the changes in the new version. The issue should always be pinned, and you can also [find the issues by searching for issues tagged with "New TypeScript Version"](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22New+TypeScript+Version%22+sort%3Acreated-desc). If the issue is open, we do not have official support yet - please be patient.
+
+In terms of what versions we support:
+
+- We do not support the `beta` releases.
+- We _generally_ do not officially support the `rc` releases.
+- We endeavour to support the latest stable TypeScript versions as soon as possible after the release.
+
+Generally we will begin working on supporting the next release when the `rc` version is released.
```

Comment:
Is this decision documented anywhere? Now that we have more maintainer funding & people, it'd be nice to be able to support RC releases. But that's really hard and I'm guessing still not something worth it.

Reply:
I'm all for us bumping versions and allowing RCs if we're able to - I just don't want us to set the expectation that we will support all RCs.

Sometimes releases are massive and and can take a lot of work to fully support.
Sometimes they have literally no impact for us.
Sometimes features from the beta get dropped from the RC.
Sometimes features from the RC get dropped from the full release.

There's just a lot of variability and with our limited bandwidth I don't want us to commit to anything!

Reply:
https://github.com/typescript-eslint/typescript-eslint/issues/5932